utils/crypto_utils.c

```diff
@@ -29,7 +30,7 @@ int get_device_id(const char *device_id) {
     }
   }
 
-  strncpy((char *)device_id, imsi, IMSI_LEN);
+  strcpy((char *)device_id, imsi);
```

Comment:
Why did you invoke `strcpy`?

Reply:
After gcc version > 8 `strncpy` is for different usage. See `-Wstringop-truncation`

Reply:
We need security advice. Can you show some reference materials?